api/envoy/extensions/load_balancing_policies/round_robin/v3/round_robin.proto

```diff
@@ -0,0 +1,26 @@
+syntax = "proto3";
+
+package envoy.extensions.load_balancing_policies.round_robin.v3;
+
+import "envoy/config/cluster/v3/cluster.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.load_balancing_policies.round_robin.v3";
+option java_outer_classname = "RoundRobinProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/load_balancing_policies/round_robin/v3;round_robinv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Round Robin Load Balancing Policy]
+
+// This configuration allows the built-in ROUND_ROBIN LB policy to be configured via the LB policy
+// extension point. See the :ref:`load balancing architecture overview
+// <arch_overview_load_balancing_types>` for more information.
+// [#extension: envoy.clusters.lb_policy]
+message RoundRobin {
+  // Configuration for slow start mode.
+  // If this configuration is not set, slow start will not be not enabled.
+  config.cluster.v3.Cluster.SlowStartConfig slow_start_config = 1;
```

Comment:
Yes, all fields need to be directly in the config for the individual LB policy extension, or else there's no way to configure them differently at different nodes in the LB policy tree.  In fact, as per discussion in #17400, the existing top-level `Cluster.common_lb_config` field won't be used with LB policies extensions; the code explicitly rejects a config that uses an LB policy extension and sets that field.

This raises the question of exactly how we want to include those fields in the LB policy extensions.  That could potentially be done just by having a `config.cluster.v3.Cluster.CommonLbConfig` field in the LB policy extension's config, but I'm not sure that actually makes the most sense; if you look at the fields in `CommonLbConfig`, it's very unlikely that any one policy would use all of them anyway.  For example, any given LB policy will use exactly one of [`ZoneAwareLbConfig` or `LocalityWeightedLbConfig`](https://github.com/envoyproxy/envoy/blob/1091423999411cbab985e0b4cd3253d8715dc71e/api/envoy/config/cluster/v3/cluster.proto#L565), so its config will never need both.  And only RING_HASH and MAGLEV will need `ConsistentHashingLbConfig`.  So ultimately, I think it will make more sense for each policy to just include the individual fields that it needs.  We can provide some common messages somewhere for reuse if it turns out to be convenient to do so.

In the case of ROUND_ROBIN, I'm not sure which if any of the `CommonLbConfig` fields are actually used, but I figured that if/when something comes up, we can add it later.

Reply:
Makes sense and I agree we can do this additively.